src/stdafx.h

```diff
@@ -81,29 +82,62 @@
 
 #include "zip.h"
 
+namespace fs = std::tr2::sys;
 #else
+#include "boost/filesystem.hpp"
+#include <stdarg.h>
+#include <unistd.h>
+namespace fs = boost::filesystem;
 
-//#include <unistd.h>
+#include <sys/syscall.h>
 
+#ifdef SYS_gettid
+pid_t gettid() { return syscall(SYS_gettid); }
+#else
+#error "SYS_gettid unavailable on this system"
 #endif
 
-namespace fs = std::tr2::sys;
+#endif
 
 #if defined(_MSC_VER)
 #define RHOST_EXPORT __declspec(dllexport)
 #define RHOST_IMPORT __declspec(dllimport)
+#define RHOST_NORETURN __declspec(noreturn)
+#define RHOST_NOTHROW __declspec(nothrow)
 #elif defined(__GNUC__)
 #define RHOST_EXPORT __attribute__((visibility("default")))
 #define RHOST_IMPORT
+#define RHOST_NORETURN
+#define RHOST_NOTHROW
 #else
 #define RHOST_EXPORT
 #define RHOST_IMPORT
+#define RHOST_NORETURN
+#define RHOST_NOTHROW
 #pragma warning Unknown DLL import/export.
 #endif
 
 #ifdef _WIN32
 #define RHOST_MAX_PATH MAX_PATH
+typedef DWORD threadid_t;
+#define RhostGetCurrentThreadId GetCurrentThreadId
+#define RHOST_TRY __try
+#define RHOST_FINALLY_START __finally {
```

Comment:
Can you remove the `{` from the macro, and also define it for non-Win32 as simply `{`? Then the braces would remain as they were at the place where it's used. And we won't need `RHOST_FINALLY_END` at all, it'll just be `}`. Since Linux doesn't have anything like SEH, these definitions should actually be correct, and not just temp stubs.

Also, can you rename this to `SEH_TRY` and `SEH_FINALLY`, to make it sure that's not for normal C++ exceptions?